### PR TITLE
GEOMESA-3011 Fix match for CLI dependency jars

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-tools/conf-filtered/geomesa-env.sh [hunk 1]
```diff
@@ -143,7 +143,7 @@ function check_classpath() {
     group="${gav%%:*}"
     artifact="${gav#$group:}"
     artifact="${artifact%%:*}"
-    if [[ $classpath != *"$artifact"* ]]; then
+    if [[ ! $classpath =~ (^|:|/)${artifact}(-[^:]*)*\.jar ]]; then
```
Comment (jnh5y) [inline geomesa-tools/conf-filtered/geomesa-env.sh:146]: Ok, what corner case does this handle?

Comment (elahrvivaz) [reply geomesa-tools/conf-filtered/geomesa-env.sh:146]: basically stricter substring matching. iirc this was to fix commons-math/commons-math3 - previously `foo.jar:commons-math3.jar` would match for artifact `commons-math`, this makes it so it doesn't.

Comment (elahrvivaz) [reply geomesa-tools/conf-filtered/geomesa-env.sh:146]: and yes, that did cause me a 10 minute headache :P

Comment (jnh5y) [reply geomesa-tools/conf-filtered/geomesa-env.sh:146]: That only took 10 minutes to figure out?  I probably would have been stuck for much longer;).

**Pull request merged**